src/lib.rs

```diff
@@ -3,6 +3,16 @@
 //!
 //! It does not do anything more than that, which makes it so fast.
 
+#![deny(rustdoc::broken_intra_doc_links)] // throw error if finds a broken link in doc
+#![warn(missing_docs // or docs are missing for public members
+        ,clippy::missing_docs_in_private_items // warn if docs are missing in private items
+        ,clippy::explicit_iter_loop // warn if you find array.iter(), &array will do the same
+        ,clippy::explicit_into_iter_loop // same with into_iter()
+        ,clippy::semicolon_if_nothing_returned // consistent semicolon endings
+        ,clippy::doc_markdown // warn if _ :: or ` are missing in .md
+        ,clippy::manual_let_else // warn if a match has only 1 arm => do if let
+)]
```

Comment:
about that, to avoid adding this kind of *Clippy* snippet to all the files, can't we just change the CI call to `clippy` and add a `toolkit.nu` with a wrapper to run *Clippy* without worrying about the options? :yum:

Reply:
i've added these rules to the GitHub workflows in 8b11060 and i've refactored the toolkit in 6dee590, is that ok with you?

the toolkit was very Nushell-oriented so i removed things related to Nushell: all should be explained in the body of 6dee590 :relieved: